x-pack/plugins/apm/server/lib/helpers/aggregated_transactions/index.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SearchAggregatedTransactionSetting } from '../../../../common/aggregated_transactions';
+import { rangeFilter } from '../../../../common/utils/range_filter';
+import { ProcessorEvent } from '../../../../common/processor_event';
+import {
+  TRANSACTION_DURATION,
+  TRANSACTION_DURATION_HISTOGRAM,
+} from '../../../../common/elasticsearch_fieldnames';
+import { APMConfig } from '../../..';
+import { APMEventClient } from '../create_es_client/create_apm_event_client';
+
+export async function getHasAggregatedTransactions({
+  start,
+  end,
+  apmEventClient,
+}: {
+  start?: number;
+  end?: number;
+  apmEventClient: APMEventClient;
+}) {
+  const response = await apmEventClient.search({
+    apm: {
+      events: [ProcessorEvent.metric],
+    },
+    body: {
+      query: {
+        bool: {
+          filter: [
+            { exists: { field: TRANSACTION_DURATION_HISTOGRAM } },
+            ...(start && end ? [{ range: rangeFilter(start, end) }] : []),
+          ],
+        },
+      },
+    },
+    terminateAfter: 1,
+  });
+
+  if (response.hits.total.value > 0) {
+    return true;
+  }
+
+  return false;
+}
+
+export async function getSearchAggregatedTransactions({
+  config,
+  start,
+  end,
+  apmEventClient,
+}: {
+  config: APMConfig;
+  start?: number;
+  end?: number;
+  apmEventClient: APMEventClient;
+}): Promise<boolean> {
+  const searchAggregatedTransactions =
+    config['xpack.apm.searchAggregatedTransactions'];
+
+  if (
+    searchAggregatedTransactions === SearchAggregatedTransactionSetting.auto
+  ) {
+    return getHasAggregatedTransactions({ start, end, apmEventClient });
+  }
+
+  return (
+    searchAggregatedTransactions === SearchAggregatedTransactionSetting.always
+  );
+}
+
+export function getTransactionDurationFieldForAggregatedTransactions(
+  searchAggregatedTransactions: boolean
+) {
+  return searchAggregatedTransactions
+    ? TRANSACTION_DURATION_HISTOGRAM
+    : TRANSACTION_DURATION;
+}
+
+export function getDocumentTypeFilterForAggregatedTransactions(
+  searchAggregatedTransactions: boolean
+) {
+  return searchAggregatedTransactions
+    ? [{ exists: { field: TRANSACTION_DURATION_HISTOGRAM } }]
+    : [];
+}
```

Comment:
This seems like a code smell: we are relying on the existence of a field to return the right type of documents. This should be handled by the processor event. Right now all metrics docs have the `metric` processor event. I think we need to have specific metric docs like `transaction_metric`.

Reply:
@sqren I very much agree (also in the context of https://github.com/elastic/kibana/issues/74498). Different document types should be managed separately. I do prefer we solve this for all metrics though (e.g. breakdown metrics should be managed separately as well). Where should we open this discussion?

Reply:
> Where should we open this discussion

Not sure. We can start the conversation with @felixbarny and @axw here or on Slack.

Reply:
Pinged them on Slack, let's chat there 👍